File: 02-programacion-basica/14-depuracion-y-pruebas/java/soluciones/RetoEsParPruebas.java
public class RetoEsParPruebas {
    public static void main(String[] args) {
        comprobar(8, true);
        comprobar(7, false);
        comprobar(0, true);
        comprobar(-4, true);
        comprobar(-3, false);
    }

    static boolean esPar(int numero) { return numero % 2 == 0; }

    static void comprobar(int numero, boolean esperado) {
        boolean obtenido = esPar(numero);
        // Un caso falla si el resultado obtenido no coincide con el esperado.
        System.out.printf("%d | esperado=%s | obtenido=%s | %s%n",
                numero, esperado, obtenido, esperado == obtenido ? "OK" : "ERROR");
    }
}

File: 02-programacion-basica/14-depuracion-y-pruebas/scala/soluciones/RetoEsParPruebas.scala
def esParReto(numero: Int): Boolean = numero % 2 == 0

@main def retoEsParPruebas(): Unit =
  // Cada tupla contiene una entrada y el resultado esperado para esa prueba.
  val casos = List(8 -> true, 7 -> false, 0 -> true, -4 -> true, -3 -> false)
  for (numero, esperado) <- casos do
    val obtenido = esParReto(numero)
    val estado = if esperado == obtenido then "OK" else "ERROR"
    println(s"$numero | esperado=$esperado | obtenido=$obtenido | $estado")
